internal/middleware/auth_Interceptor.go

```diff
@@ -0,0 +1,141 @@
+package middleware
+
+import (
+	"github.com/google/uuid"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+	"time"
+)
+
+// UnaryInterceptor - гRPC интерсептор для логирования и проверки авторизации.
+func (a *AuthMiddleware) UnaryInterceptor(ctx context.Context,
+	req interface{},
+	info *grpc.UnaryServerInfo,
+	handler grpc.UnaryHandler) (resp interface{}, err error) {
+
+	// Логируем запрос с использованием вашего логгера
+	start := time.Now()
+	a.log.Info(
+		"gRPC request started",
+		"method", info.FullMethod,
+		"request", req,
+	)
+
+	// Методы, которые требуют CheckAuthInterceptor
+	methodsRequiringCheckAuth := map[string]bool{
+		"/shortener.ShortenerService/GetUsersURLs": true,
+		"/shortener.ShortenerService/DeletionURLs": true,
+		"/shortener.ShortenerService/GetStatus":    true,
+	}
+
+	// Если метод в списке, применяем CheckAuthInterceptor
+	if methodsRequiringCheckAuth[info.FullMethod] {
+		resp, err = a.UnaryCheckAuthInterceptor(ctx, req, info, handler)
+	} else {
+		// Для всех остальных методов применяем AuthInterceptor
+		resp, err = a.UnaryAuthInterceptor(ctx, req, info, handler)
+	}
+
+	// Логируем завершение запроса с использованием логгера
+	duration := time.Since(start)
+	if err != nil {
+		// Логируем ошибку
+		a.log.Error(
+			"gRPC request failed",
+			"method", info.FullMethod,
+			"error", err,
+			"duration", duration,
+		)
+	} else {
+		// Логируем успешный ответ
+		a.log.Info(
+			"gRPC request completed",
+			"method", info.FullMethod,
+			"response", resp,
+			"duration", duration,
+		)
+	}
+
+	// Возвращаем ответ и ошибку
+	return resp, err
```

Comment:
Я бы предложил явно вернуть ошибку или `nil`. Это немного избыточно с точки зрения кол-ва `return`, но намного более читаемо.
 
```suggestion
	if err != nil {
		// Логируем ошибку
		a.log.Error(
			"gRPC request failed",
			"method", info.FullMethod,
			"error", err,
			"duration", duration,
		)
		return resp, err
	}

	// Логируем успешный ответ
	a.log.Info(
		"gRPC request completed",
		"method", info.FullMethod,
		"response", resp,
		"duration", duration,
	)

	// Возвращаем ответ и ошибку
	return resp, nil
```

Reply:
исправил